src/phylo/iteration.jl

```diff
@@ -0,0 +1,277 @@
+#==================================#
+# Iteration and traversal of trees #
+#==================================#
+
+# Ben J. Ward, 2014.
+
+@Docile.doc """
+`PhylogenyIterator` is an abstract type that defines a family of iterators
+for traversing trees in various ways, including `BreadthFirst`, and `DepthFirst`,
+or from a tip to a root.
+""" ->
+abstract PhylogenyIterator
```

Comment:
The main reason it is used is for Parameteric definition of the Base.search and searchall functions. But maybe there's something better we can do.